src/gsCore/gsMath.h

```diff
@@ -321,6 +321,14 @@ template <typename T> int getSign(T val)
     return (T(0) < val) - (val < T(0));
 }
 
+/// Return smallest difference between two (integer) numbers as size_t
+/// especially the case abs_diff(INT32_MAX, INT32_MIN) := UINT32_MAX is correct
+template <typename T>
+inline typename std::make_unsigned<T>::type abs_diff(T a, T b)
+{
+    return a > b ? a - b : b - a;
```

Comment:
i think, one has to cast a and b first to the resulting type

a > b ? static_cast<...>(a) - static_cast<...>(b) : static_cast<...>(b) - static_cast<...>(a)

Reply:
to avoid overflow!?

Reply:
Yes, https://timsong-cpp.github.io/cppwp/n4659/conv.integral allows it